### PR TITLE
[Helm] Correction in templates and enabling use of PowerMode and Ebpf

### PR DESCRIPTION
**Pull request by jullianow**
After checking the latest version present on marter, it was possible to identify that some features were not yet ready for use.
That is, they were not being set correctly/as expected under some conditions.
Below I list what was found and fixed.

1. Helper to set Service Account Name:
The service account is only created when the `rbac.create` property is enabled.
The problem identified was that when the service account is not created, the Cluster Role Binding was referencing a service account that necessarily has to have the same name as the chart. Which is not true when reading the documentation (since it should get the name from the `rbac.serviceAccountName` property).
So, to solve this problem, a function was created in a single place with the name of the service account, so that it is possible to use the return of this function where it makes sense (`ClusterRoleBinding`, `Deployment`, `Daemonset`)

2. Created a function to handle all settings used by `DaemonSet` (other than received from a `ConfigMap` (`config-args` key))

3. `low_power` configuration was only being injected when Secret is created by chart. However, in the case where the `Secret` is created outside the helm, this property was not defined. So, to make it easier, was defined a function that receives the configuration values referring to the `DaemonSet` in just one place.

4. The `enable_bpf_sensors` property was not being set for the `tsagent`, even though the chart already got a property for it. Such a configuration has been made available.

5. Existing conditions in Secret and Service Account were starting and ending with hyfem (E.g: `{{- if .Values.rbac.create -}}` and  `{{- if not .Values.agentSetupExternalSecretRef -}}`). In this way, a line after the condition was being removed, causing the first line to be in the commented area, as can be seen below.

image
image

### CODE REVIEW

**Comment by jullianow** [comment]
@olhado These adjustments avoid impacts on a new version of the chart.
I'm looking forward to a new version to make use of LowPowerMode and Ebpf, to improve the visibility of my environments.
When do you plan to release a new version of the chart? Just to schedule me internally for a new release.

**Comment by olhado** [comment]
Hi @jullianow

I was informed after accepting your previous PR, that we have a new process to accept changes, which I believe are in the readme in master. This is something new for external contributors since our acquisition, so until you complete the process to be a contributor, I can’t accept these changes.

That said, if you do complete the contributor signup process, we can review this PR more. Most of it looks good, but some of your comments seemed to not have a corresponding change, or missed some of the new changes we have committed to master since our last official release. But I could be the one misunderstanding 😃

I am going to leave this PR open in the hopes that you complete the contributor process, and we can get this in at some point.

**Comment by olhado** [comment]
Here is the new contributing doc: https://github.com/threatstack/threatstack-helm/blob/master/CONTRIBUTING.md

I apologize for the confusion, this is a new process for me too. And I notice that the document doesn’t have info on what needs to be sent, so if you are still interested, I can figure out on our side how to make it happen.

**Comment by jullianow** [comment]
Okay @olhado .
It was also unclear to me what I need to do to be able to contribute. If possible, get more information for me.

Thanks

**Comment by olhado** [comment]
@jullianow Just adding this comment to let you know we haven't forgotten, there's just travel/vacation and the like on our end slowing it down.

**Comment by jullianow** [comment]
@olhado I just sent the signed document to the email mentioned in the document.

**Comment by patcable** [comment]
Hi @jullianow -- I havent seen that completed form come into `aipo-cla@f5.com` and it is allegedly not in our spam filter. Mind sending it again?

**Comment by jullianow** [comment]
@patcable 
image

**Comment by patcable** [comment]
Got it that time. Thanks!

**Review by olhado — APPROVED** [review]
Going to fix in a separate PR before release.

**Comment by olhado** [comment]
Going to merge in the fixes, thanks @jullianow !

File: templates/secrets.yaml [hunk 1]
```diff
@@ -32,5 +32,5 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 type: Opaque
 stringData:
-  ts-setup-args: "--deploy-key {{ .Values.agentDeployKey }} --ruleset '{{ .Values.rulesets }}' {{ include "threatstack-agent.daemonset-lowpower-config" . }} {{ .Values.additionalSetupConfig }}"
-{{- end -}}
+  ts-setup-args: "--deploy-key {{ .Values.agentDeployKey }} --ruleset '{{ .Values.rulesets }}'"
```
[the remark this answers is not in the document]
Comment (jullianow) [reply templates/secrets.yaml:35]: Ok

**Pull request merged**